File: components/SSD.java
package components; // ← Fará parte do pacote "components", junto com outras peças de computador.
// ↓ Criando a classe SSD (memória secundária do tipo "estado sólido")
public class SSD {
 private String nome;
 private Integer gb;
 private String conn;
 // ↓ Argumentos para criar a classe
 public SSD (String nome, Integer gb, String conn) {
  // ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if (nome == null || nome.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  if (gb <= 0) {throw new IllegalArgumentException("Deve ser maior que zero!");}
  if (conn == null || conn.trim().isEmpty()) {throw new IllegalArgumentException("Não pode ser vazio ou nulo!");}
  this.nome=nome;
  this.gb=gb;
  this.conn=conn;
 }
 // ↓ Simplesmente digitando o nome do instance dentro do parênteses no "System.out.println()" irá renderizar a string abaixo
 @Override
 public String toString() {
  return "\n::::: PROPRIEDADES DA MEMÓRIA SECUNDÁRIA :::::\nNome: "+nome+"\nCapacidade: "+gb+" GB\nConexão: "+conn+"\n";
 }
 // ↓ NOTA: Para manter os atributos da classe como private e para serem visíveis por outras classes, os getters abaixo serão utilizados.
 public String verNome() {return nome;}
 public String verConexao() {return conn;}
 public Integer verGB() {return gb;}
}

File: components/SSD.py
# ↓ Criando a classe SSD (memória secundária do tipo "estado sólido")
class SSD:
 lista_conn=["SATA","IDE","NVME"]
 # ↓ Argumentos para criar a classe
 def __init__(self, nome:str, gb:int, conn:str):
  
  # ↓ NOTA: Os tratadores de exceção abaixo, marcados com if, são personalizados.
  if nome is None or nome.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  if gb <= 0:
   raise ValueError("Deve ser maior que zero!")
  if conn is None or conn.strip() == "":
   raise ValueError("Não pode ser vazio ou nulo!")
  if conn.upper() not in SSD.lista_conn:
   raise TypeError("Conexão de armazenamento não está na lista das válidas!")
  
  self.__nome=nome
  self.__gb=gb
  self.__conn=conn
  
  # ↓ Simplesmente digitando o nome do instance no console, ou dentro do parênteses no "print()" irá renderizar a string abaixo
 def __repr__(self) -> str:
  return f"\n::::: PROPRIEDADES DA MEMÓRIA SECUNDÁRIA :::::\nNome: {self.__nome}\nCapacidade: {self.__gb} GB\nConexão: {self.__conn}\n"
  
  # ↓ NOTA: Para manter os atributos da classe como privados e para serem visíveis por outras classes, os getters abaixo serão utilizados.
 def verNome(self) -> str:
  return self.__nome
 def verGB(self) -> int:
  return self.__gb
 def verConexao(self) -> str:
  return self.__conn
